### PR TITLE
Issue497 carnot

### LINKED ISSUE #497: Default values for Carnot chiller/heat pumps can violate 2nd law
The `Carnot_*` heat pumps and chillers use as the default values for computing the Carnot effectiveness inlet and outlet temperatures for the condenser or evaporator.

At small temperature lifts, using inlet temperatures only can lead to a computation of the Carnot efficiency that is higher than the actual Carnot efficiency.

In a model, we observed that for the situation of using the inlet temperatures _and_ having low lifts, the entropy generated by the chiller was negative, which is a violation of the 2nd law. Therefore, the default values should be changed to using the outlet temperatures, which is a more conservative assumption that leads to lower Carnot efficiencies, and ensures that the 2nd law is not violated.

### PR DESCRIPTION
**Pull request by Mathadon**
this closes #497 

I ran unit tests for Annex60.Fluid.Chillers.

Some unit tests give large differences for the COP:

screen shot 2017-01-02 at 15 24 26

and a non-linear algebraic loop of size 1 is generated, which makes sense since `energyDynamics` are `SteadyState` by default.

### CODE REVIEW

**Comment by mwetter** [comment]
@Mathadon 
I merged the previous development branch with your new branch, updated the `Fluid.HeatPumps` models (which did no longer translate) and run all regression tests.

Can you please review the new files that I added (examples and entropy sensors) and then merge to the master and delete the two development branches.

File: Annex60/Fluid/Chillers/Carnot_TEva.mo [hunk 1]
```diff
@@ -165,6 +165,15 @@ This is for
 issue 497</a>.
 </li>
 <li>
+August 8, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/Chillers/Carnot_TEva.mo]: It seems that you forgot to update the dates. I will fix this.

File: Annex60/Fluid/Chillers/Carnot_y.mo [hunk 2]
```diff
@@ -94,6 +94,15 @@ This is for
 issue 497</a>.
 </li>
 <li>
+August 8, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/Chillers/Carnot_y.mo]: date

File: Annex60/Fluid/Chillers/Validation/Carnot_TEva_2ndLaw.mo [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+<p>
+The example consists of two identical models, except that the instance
+<code>chi_a</code> uses the temperatures at the inlets to compute the
+Carnot efficiency, whereas the instance <code>chi_b</code> uses
+the outlet temperatures.
+</p>
+</html>", revisions="<html>
+<ul>
+<li>
+November 18, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/Chillers/Validation/Carnot_TEva_2ndLaw.mo]: date

File: Annex60/Fluid/HeatPumps/Carnot_TCon.mo [hunk 4]
```diff
@@ -166,6 +164,15 @@ This is for
 issue 497</a>.
 </li>
 <li>
+August 8, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/HeatPumps/Carnot_TCon.mo]: date

File: Annex60/Fluid/HeatPumps/Carnot_y.mo [hunk 5]
```diff
@@ -95,6 +95,15 @@ This is for
 issue 497</a>.
 </li>
 <li>
+August 8, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/HeatPumps/Carnot_y.mo]: date

File: Annex60/Fluid/Sensors/Examples/EntropyFlowRate.mo [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+<p>
+This example tests the entropy flow rate sensor and the
+specific entropy sensor.
+The model compares the output of the entropy flow rate sensor with
+the product of the output of the entropy and the mass flow rate sensor.
+</p>
+</html>", revisions="<html>
+<ul>
+<li>
+July 29, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/Sensors/Examples/EntropyFlowRate.mo]: date

File: Annex60/Fluid/Sensors/EntropyFlowRate.mo [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+<i>S&#775; = m&#775; s<sub>out</sub></i>
+is computed using a first order differential equation.
+See <a href=\"modelica://Annex60.Fluid.Sensors.UsersGuide\">
+Annex60.Fluid.Sensors.UsersGuide</a> for an explanation.
+</p>
+</html>",
+revisions="<html>
+<ul>
+<li>
+July 29, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/Sensors/EntropyFlowRate.mo]: date

File: Annex60/Fluid/Chillers/Carnot_TEva.mo [hunk 8]
```diff
@@ -160,6 +156,24 @@ Annex60.Fluid.HeatPumps.Examples.Carnot_TCon</a>.
...
+January 2, 2017, by Filip Jorissen:<br/>
+Removed parameters
+<code>effInpEva</code> and <code>effInpCon</code>
+and updated documentation.
+This is for
+<a href=\"https://github.com/iea-annex60/modelica-annex60/issues/497\">
+issue 497</a>.
+</li>
+<li>
+August 8, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/Chillers/Carnot_TEva.mo]: I am a bit confused whether this date is intentional chosen or if you forgot to edit it?

File: Annex60/Fluid/Chillers/Carnot_y.mo [hunk 9]
```diff
@@ -90,6 +85,24 @@ For a similar model that can be used as a heat pump, see
...
+January 2, 2017, by Filip Jorissen:<br/>
+Removed parameters
+<code>effInpEva</code> and <code>effInpCon</code>
+and updated documentation.
+This is for
+<a href=\"https://github.com/iea-annex60/modelica-annex60/issues/497\">
+issue 497</a>.
+</li>
+<li>
+August 8, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/Chillers/Carnot_y.mo]: I am a bit confused whether this date is intentional chosen or if you forgot to edit it?

File: Annex60/Fluid/HeatPumps/Carnot_TCon.mo [hunk 10]
```diff
@@ -160,6 +155,24 @@ Annex60.Fluid.Chillers.Examples.Carnot_TEva</a>.
...
+January 3, 2017, by Michael Wetter:<br/>
+Removed parameters
+<code>effInpEva</code> and <code>effInpCon</code>
+and updated documentation.
+This is for
+<a href=\"https://github.com/iea-annex60/modelica-annex60/issues/497\">
+issue 497</a>.
+</li>
+<li>
+August 8, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/HeatPumps/Carnot_TCon.mo]: I am a bit confused whether this date is intentional chosen or if you forgot to edit it?

File: Annex60/Fluid/HeatPumps/Carnot_y.mo [hunk 11]
```diff
@@ -91,6 +86,24 @@ For a similar model that can be used as a chiller, see
...
+January 3, 2017, by Michael Wetter:<br/>
+Removed parameters
+<code>effInpEva</code> and <code>effInpCon</code>
+and updated documentation.
+This is for
+<a href=\"https://github.com/iea-annex60/modelica-annex60/issues/497\">
+issue 497</a>.
+</li>
+<li>
+August 8, 2016, by Michael Wetter:<br/>
```
Comment (Mathadon) [inline Annex60/Fluid/HeatPumps/Carnot_y.mo]: I am a bit confused whether this date is intentional chosen or if you forgot to edit it?

File: Annex60/Fluid/Chillers/Validation/Carnot_TEva_2ndLaw.mo [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+    annotation (Placement(transformation(extent={{-40,50},{-20,70}})));
+  Modelica.Blocks.Sources.Ramp dTConEva(
+    duration=1,
+    offset=25,
+    height=-25)
+               "Temperature lift condensor inlet minus evaporator outlet"
+    annotation (Placement(transformation(extent={{-80,0},{-60,20}})));
+  Modelica.Blocks.Math.Add TConIn "Condensor inlet temperature"
+    annotation (Placement(transformation(extent={{0,6},{20,26}})));
+  chiller chi_b(
```
Comment (Mathadon) [inline Annex60/Fluid/Chillers/Validation/Carnot_TEva_2ndLaw.mo]: Should this class not have a capital c in its declaration?

File: Annex60/Fluid/Chillers/Validation/Carnot_TEva_2ndLaw.mo [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+    parameter Modelica.SIunits.TemperatureDifference dTCon_nominal
+      "Temperature difference condenser outlet-inlet";
+    parameter Modelica.SIunits.HeatFlowRate QEva_flow_nominal
+      "Evaporator heat flow rate";
+    parameter Modelica.SIunits.MassFlowRate m1_flow_nominal
+      "Nominal mass flow rate at condeser water side";
+    parameter Modelica.SIunits.MassFlowRate m2_flow_nominal
+      "Nominal mass flow rate at chilled water side";
+
+    Sources.MassFlowSource_T sou1(
```
Comment (Mathadon) [inline Annex60/Fluid/Chillers/Validation/Carnot_TEva_2ndLaw.mo]: The chiller model does not show the full path of the instance. I'll fix this.

File: Annex60/Fluid/Sensors/EntropyFlowRate.mo (hunk 7 above)
Comment (mwetter) [inline Annex60/Fluid/Sensors/EntropyFlowRate.mo]: This date is correct:
```
$ git log Annex60/Fluid/Sensors/Examples/EntropyFlowRate.mo
...
commit 720ac79a9dbeae3f09b907c4581ff98269f7ce14
Author: Michael Wetter <MWetter@lbl.gov>
Date:   Fri Jul 29 23:02:14 2016 -0700

    Added entropy flow rate sensor
```

File: Annex60/Fluid/Chillers/Carnot_y.mo (hunk 9 above)
Comment (mwetter) [inline Annex60/Fluid/Chillers/Carnot_y.mo]: The date is correct.

File: Annex60/Fluid/HeatPumps/Carnot_TCon.mo (hunk 10 above)
Comment (mwetter) [inline Annex60/Fluid/HeatPumps/Carnot_TCon.mo]: The date is correc

File: Annex60/Fluid/HeatPumps/Carnot_y.mo (hunk 11 above)
Comment (mwetter) [inline Annex60/Fluid/HeatPumps/Carnot_y.mo]: The date is correct.

File: Annex60/Fluid/Chillers/Validation/Carnot_TEva_2ndLaw.mo (hunk 12 above)
Comment (mwetter) [inline Annex60/Fluid/Chillers/Validation/Carnot_TEva_2ndLaw.mo]: Thanks, I renamed it in e9757e0

**Pull request merged**